src/App.tsx

```diff
@@ -15,37 +17,98 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App: React.FC = () => {
+enum SortType {
+  sort_field_alphabet = 'alphabet',
+  sort_field_length = 'length',
+  sort_field_none = '',
+}
+
+function getPreparedGoods(
+  goods: string[],
+  sortField: SortType,
+  isReversed: boolean,
+): string[] {
+  const preparedGoods = [...goods];
+
+  switch (sortField) {
+    case SortType.sort_field_alphabet:
+      preparedGoods.sort((good1: string, good2: string) =>
+        good1.localeCompare(good2),
+      );
+      break;
+
+    case SortType.sort_field_length:
+      preparedGoods.sort(
+        (good1: string, good2: string) => good1.length - good2.length,
+      );
+      break;
+
+    default:
+      break;
+  }
+
+  if (isReversed) {
+    preparedGoods.reverse();
+  }
+
+  return preparedGoods;
+}
+
+export const App = () => {
+  const [sortField, setSortField] = useState<SortType>(
+    SortType.sort_field_none,
+  );
+  const [isReversed, setIsReversed] = useState(false);
+
+  const visibleGoods = getPreparedGoods(goodsFromServer, sortField, isReversed);
+  const showResetButton = sortField || isReversed;
+
   return (
     <div className="section content">
       <div className="buttons">
-        <button type="button" className="button is-info is-light">
+        <button
+          type="button"
+          className={classNames('button is-info', {
+            'is-light': sortField !== SortType.sort_field_alphabet,
+          })}
+          onClick={() => setSortField(SortType.sort_field_alphabet)}
+        >
           Sort alphabetically
         </button>
 
-        <button type="button" className="button is-success is-light">
+        <button
+          type="button"
+          className={classNames('button is-info', {
+            'is-light': sortField !== SortType.sort_field_length,
+          })}
+          onClick={() => setSortField(SortType.sort_field_length)}
+        >
           Sort by length
         </button>
 
-        <button type="button" className="button is-warning is-light">
+        <button
+          type="button"
+          className={`button is-info ${isReversed ? '' : 'is-light'}`}
```

Comment:
use classNames here too

Also, in the working example this button isn't blue (not is-info class)